discovery.rst

```diff
@@ -62,8 +87,8 @@ Common Errors
 Puli Factory is not available
```

Comment:
I think it is good as it is. 

But we should add docs about the "DiscoveryFailedException" and maybe rewrite the "NotFoundException" a bit.